### PR TITLE
Get recipe branch

### LINKED ISSUE #17: Create Get a Recipe Query
**Function / Endpoint name: getRecipe**
- The recipe id will need to be sent in the request from FE

Data structure: 

```
{
 id: 'string',
 recipeName: 'string',
 recipeAuthor: 'string',
 description: 'sting', /* option to be empty*/
 instructions: 'string',
 ingredients: [
   {
     name: 'string',
     unit: 'string', /* option to be empty*/
     measurement: 0,
   },
   {
     name: 'sting',
     unit: 'sting', /* option to be empty*/
     measurement: 0,
   },
 ]
} 
```

### PR DESCRIPTION
**Pull request by mlynch5187**
### Type of change made:
- [x] New feature (non-breaking change which adds functionality)
- [ ] Testing - building tests
- [ ] Formatting - no new features
- [ ] Styling- no new features
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
### Description
* This branch adds recipe query functionality. We can now query all recipes, or a specific recipe.
### How should this be manually tested?
* Run the phoenix server (mix phx.server) in your terminal and navigate to localhost:4000 in your browser. Here you can test queries.
### Any background context you want to provide?
* This was done during a 24 hour solo spike. Each team member had specific queries or mutations assigned to us.
### What are the relevant tickets?
* Resolves #17 #27 
### Checklist
- [ ] I have tested this code
- [ ] I have updated the Readme
### Questions:
None

### CODE REVIEW

File: lib/roots/recipe.ex [hunk 1]
```diff
@@ -13,6 +13,10 @@ defmodule Roots.Recipe do
     timestamps()
   end
 
+  def all do
```
Comment (RyN21) [inline lib/roots/recipe.ex:16]: Nice looks good!

File: lib/roots_web/resolvers/recipe_resolver.ex [hunk 2]
```diff
@@ -0,0 +1,14 @@
+defmodule RootsWeb.Resolvers.RecipeResolver do
```
Comment (RyN21) [inline lib/roots_web/resolvers/recipe_resolver.ex:1]: Resolvers lookin great

File: lib/roots_web/schema/recipe_types.ex [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  @desc "Recipe"
+  object :recipe do
+    field :id, :id
+    field :description, :string
+    field :instructions, :string
+    field :title, :string
+    field :author, :string
+
+    field :cookbook, :cookbook, resolve: dataloader(Data)
```
Comment (RyN21) [inline lib/roots_web/schema/recipe_types.ex:16]: Recipe types file looks good!
One thing I think we can do now is uncomment line 15 in cookbook_types to tie the relationships between recipe and cookbook together now. Same line as you have on 17 for ingredients

**Review by RyN21 — DISMISSED** [review]

**Review by RyN21 — CHANGES_REQUESTED** [review]
Just one request. In shema/cookbook_types, we can uncomment "field :recipes, list_of(:recipe), resolve: dataloader(Data)" I believe on line 15, so we can tie the relationships together now that you added recipe

**Review by RyN21 — APPROVED** [review]

File: priv/repo/seeds.exs [hunk 4]
```diff
@@ -9,3 +9,100 @@
 #
 # We recommend using the bang functions (`insert!`, `update!`
 # and so on) as they will fail if something goes wrong.
+alias Roots.Repo
```
Comment (ejdelsztejn) [inline priv/repo/seeds.exs:12]: This seed file looks awesome!

**Review by ejdelsztejn — APPROVED** [review]

File: lib/roots_web/schema/recipe_types.ex (hunk 3 above)
Comment (mlynch5187) [reply lib/roots_web/schema/recipe_types.ex:16]: Got it - I'll add the relationship between recipe and cookbook, and re-request your review.

**Pull request merged**